hera_cal/abscal_funcs.py

```diff
@@ -1267,28 +1344,53 @@ def avg_data_across_red_bls(data, bls, antpos, flags=None, broadcast_flags=True,
         else:
             d = np.nanmean(map(lambda k: data[k], bl_group), axis=0)
 
-        # assign to red_data
+        # get flags
+        if broadcast_flags:
+            f = np.max(map(lambda k: flags[k], bl_group), axis=0).astype(np.bool)
+        else:
+            f = np.min(map(lambda k: flags[k], bl_group), axis=0).astype(np.bool)
+
+        # iterate over bl_group
         for j, key in enumerate(sorted(bl_group)):
-            red_data[key] = copy.copy(d)
+            # assign to red_data and flags
+            red_data[key] = d
+            red_flags[key] = f
+
+            # break if no mirror
             if mirror_red_data is False:
                 break
 
-        # assign flags
-        if broadcast_flags:
-            red_flags[key] = np.max(map(lambda k: flags[k], bl_group), axis=0).astype(np.bool)
-        else:
-            red_flags[key] = np.min(map(lambda k: flags[k], bl_group), axis=0).astype(np.bool)
-
     # get red_data keys
     red_keys = red_data.keys()
 
-    return red_data, red_flags, red_keys
+    return DataContainer(red_data), DataContainer(red_flags), red_keys
 
 
 def avg_file_across_red_bls(data_fname, outdir=None, output_fname=None,
                             write_miriad=True, output_data=False, overwrite=False, 
                             verbose=True, **kwargs):
     """
+    Open a file and run avg_data_across_red_bls on data, then write to file
+
+    Parameters:
+    -----------
+    data_fname : type=str, path to miriad file
+
+    outdir : type=str, output directory
```

Comment:
If None, does it just not write?

Reply:
if none, it uses the current working directory